### PR TITLE
imp: add bech32 address support for Ethermint chains 

### LINKED ISSUE #211: Bech32 address derivation failing on Ethermint chains
Current implementation fails to parse wallet address to Bech32 correctly for Evmos chain

Steps to reproduce:

- `config.toml`
```
chain_id = "evmos_9000-1"
validators = 3
accounts = 2
binary = "evmosd"
hrp_prefix = "evmos"
denom = "aevmos"
coin_type = 60
account_balance = 100000000000000000000000000
validator_balance = 100000000000000000000000000
verbose = true

[config_node.app]
minimum-gas-prices = "0aevmos"
api = { enable = true, swagger = true, enabled-unsafe-cors = true }
rosetta = { enable = false }

[config_node.config]
instrumentation = { prometheus = false }
p2p = { addr_book_strict = false, allow_duplicate_ip = true }

[config_genesis.app_state.gov]
voting_params = { voting_period = "30s" }

[config_genesis.app_state.gov.deposit_params]
min_deposit = [{ denom = "aevmos", amount = "10000000" }]
max_deposit_period = "30s"

[config_genesis.app_state.staking.params]
bond_denom = "aevmos"

[config_genesis.app_state.claims.params]
claims_denom = "aevmos"

[config_genesis.app_state.evm.params]
evm_denom = "aevmos"

[config_genesis.app_state.crisis.constant_fee]
denom = "aevmos"

[config_genesis.consensus_params.block]
max_gas = "10000000"

[config_node.config.consensus]
timeout_commit = "100ms"
```
- Run `atomkraft chain testnet`

Will get error that address does not exist on genesis file

**Proposed solution**
Use the wallet address bytes for the Bech32 encoding

```
bip_utils.Bech32Encoder.Encode(prefix, bytes.fromhex(self.wallet.address()[2:]))
```

### PR DESCRIPTION
**Pull request by GAtom22**
## Description

This PR introduces the changes needed to make the proper bech32 address encoding to be able to use Atomkraft with Ethermint chains

Closes https://github.com/informalsystems/atomkraft/issues/211

### CODE REVIEW

File: atomkraft/chain/node.py [hunk 1]
```diff
@@ -72,11 +78,13 @@ def __init__(
             np.random.default_rng(list(bytes(final_seed))).bytes(strength // 8).hex()
         )
         self.coin_type = coin_type
+        coin = supported_coins.setdefault(self.coin_type)
```
Comment (rnbguy) [inline atomkraft/chain/node.py]: why do you call `.setdefault()`? isn't `.get()` enough?

File: atomkraft/chain/node.py [hunk 2]
```diff
@@ -72,11 +78,13 @@ def __init__(
...
+        if coin is None:
+            raise ValueError(
+                f"Provided coin type {self.coin_type} not supported."
+            )
 
-        class LocalCC(Cryptocurrency):
-            COIN_TYPE = CoinType({"INDEX": self.coin_type, "HARDENED": True})
-
-        self.wallet = hdwallet.BIP44HDWallet(cryptocurrency=LocalCC).from_entropy(
+        self.wallet = hdwallet.BIP44HDWallet(cryptocurrency=coin).from_entropy(
```
Comment (rnbguy) [inline atomkraft/chain/node.py]: I can't allow this change, as Cosmos community uses coin types others than 118.

Example,

| SLIP44 | Symbol | Chain         |
|--------|--------|---------------|
| 529    | SCRT   | Secret Network|
| 394    | CRO    | Crypto Chain  |
| 330    | LUNA   | Terra         |
| 852    | DSM    | Desmos        |

Comment (rnbguy) [reply atomkraft/chain/node.py]: maybe, add a logic, when coin is None, it falls back to `LocalCC`.

File: atomkraft/chain/node.py [hunk 3]
```diff
@@ -91,7 +99,11 @@ def validator_address(self, prefix) -> str:
         return self.bech32_address(f"{prefix}valoper")
 
     def bech32_address(self, prefix) -> str:
-        return bip_utils.Bech32Encoder.Encode(prefix, bytes.fromhex(self.wallet.hash()))
+        match self.coin_type:
+            case EthereumMainnet.COIN_TYPE.INDEX:
+                return bip_utils.Bech32Encoder.Encode(prefix, bytes.fromhex(self.wallet.address()[2:]))
```
Comment (rnbguy) [inline atomkraft/chain/node.py:101]: I know this is correct, as I just tested it with `evmosd key parse`. But can you provide an official reference for this?

**Review by rnbguy — CHANGES_REQUESTED** [review]
Many thanks for this PR. I requested some changes. Once resolved, I will merge it.

File: atomkraft/chain/node.py (hunk 3 above)
Comment (GAtom22) [reply atomkraft/chain/node.py:101]: By using the address bytes for address encoding, we can convert addresses between hex and bech32 easily. See [this cmd](https://github.com/evmos/ethermint/blob/2814c7857f1cdbb35cdad21bf3bc160a6b29021b/client/debug/debug.go#L83-L116) that is used for this purpose

**Review by rnbguy — APPROVED** [review]
LGTM :+1:

**Comment by rnbguy** [comment]
Thanks @GAtom22 :pray:

**Pull request merged**